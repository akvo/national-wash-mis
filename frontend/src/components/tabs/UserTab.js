import React from "react";
import { Tabs } from "antd";
import { useNavigate } from "react-router-dom";

const { TabPane } = Tabs;

const UserTab = ({ tabBarExtraContent }) => {
  const pathname = window.location.pathname;
  const navigate = useNavigate();

  return (
    <Tabs
      size="large"
      activeKey={pathname}
      onChange={(key) => navigate(key)}
      tabBarExtraContent={tabBarExtraContent}
    >
      <TabPane tab="Manage Users" key="/users" data-testid="manage-users">
        &nbsp;
      </TabPane>
      <TabPane
        tab="Data Validation Tree"
        key="/approvers/tree"
        data-testid="manage-data-validation"
      >
        &nbsp;
      </TabPane>
    </Tabs>
  );
};

export default UserTab;
