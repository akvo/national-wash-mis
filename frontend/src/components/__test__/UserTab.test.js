import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Link, MemoryRouter } from "react-router-dom";
import UserTab from "../tabs/UserTab";
import { Button } from "antd";

describe("components/UserTab", () => {
  test("test if UserTab has two tabs and add user", () => {
    const { container } = render(
      <UserTab
        tabBarExtraContent={
          <Link to="/user/add">
            <Button type="primary">Add new user</Button>
          </Link>
        }
      />,
      { wrapper: MemoryRouter }
    );
    const addUser = container.querySelector(`[href="/user/add"]`);
    expect(screen.getByText("Manage Users")).toBeInTheDocument();
    expect(addUser).toHaveAttribute("href", "/user/add");
    expect(container).toMatchSnapshot();
  });
});
