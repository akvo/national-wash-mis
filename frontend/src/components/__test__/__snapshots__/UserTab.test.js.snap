// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/UserTab test if UserTab has two tabs and add user 1`] = `
<div>
  <div
    class="ant-tabs ant-tabs-top ant-tabs-large"
  >
    <div
      class="ant-tabs-nav"
      role="tablist"
    >
      <div
        class="ant-tabs-nav-wrap"
      >
        <div
          class="ant-tabs-nav-list"
          style="transform: translate(0px, 0px);"
        >
          <div
            class="ant-tabs-tab"
          >
            <div
              aria-controls="rc-tabs-test-panel-/users"
              aria-selected="false"
              class="ant-tabs-tab-btn"
              id="rc-tabs-test-tab-/users"
              role="tab"
              tabindex="0"
            >
              Manage Users
            </div>
          </div>
          <div
            class="ant-tabs-tab"
          >
            <div
              aria-controls="rc-tabs-test-panel-/approvers/tree"
              aria-selected="false"
              class="ant-tabs-tab-btn"
              id="rc-tabs-test-tab-/approvers/tree"
              role="tab"
              tabindex="0"
            >
              Data Validation Tree
            </div>
          </div>
          <div
            class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
          />
        </div>
      </div>
      <div
        class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
      >
        <button
          aria-controls="rc-tabs-test-more-popup"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-hidden="true"
          class="ant-tabs-nav-more"
          id="rc-tabs-test-more"
          style="visibility: hidden; order: 1;"
          tabindex="-1"
          type="button"
        >
          <span
            aria-label="ellipsis"
            class="anticon anticon-ellipsis"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="ellipsis"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        class="ant-tabs-extra-content"
      >
        <a
          href="/user/add"
        >
          <button
            class="ant-btn ant-btn-primary"
            type="button"
          >
            <span>
              Add new user
            </span>
          </button>
        </a>
      </div>
    </div>
    <div
      class="ant-tabs-content-holder"
    >
      <div
        class="ant-tabs-content ant-tabs-content-top"
      >
        <div
          aria-hidden="true"
          aria-labelledby="rc-tabs-test-tab-/users"
          class="ant-tabs-tabpane"
          id="rc-tabs-test-panel-/users"
          role="tabpanel"
          style="display: none;"
          tabindex="-1"
        />
        <div
          aria-hidden="true"
          aria-labelledby="rc-tabs-test-tab-/approvers/tree"
          class="ant-tabs-tabpane"
          id="rc-tabs-test-panel-/approvers/tree"
          role="tabpanel"
          style="display: none;"
          tabindex="-1"
        />
      </div>
    </div>
  </div>
</div>
`;
